### PR TITLE
New street addr. georeference tab

### LINKED ISSUE #4158: New georeference modal
This issue is to coordinate all the work related to the new georeference modal. 
Design proposals: [#3828](https://github.com/CartoDB/cartodb/issues/3828)

- New base modal w/ the simplest use-case (lon/lat tab) #4192
  - [x] new structure
  - [x] lon/lat tab implemented
  - [x] can be opened from the options menu, header, and the "_no georeferenced data on your layer_"
- the other tabs
  - [x] city names tab #4222
  - [x] admin regions #4222
  - [x] postal codes #4222
  - [x] IP address #4255
  - [x] street addresses #4316 
- special cases: #4255
  - [x] [disable all but lon/lat georeference tabs]([screen shot 2015-06-29 at 16 35 21](https://cloud.githubusercontent.com/assets/978461/8413966/d053cd92-1e94-11e5-90b4-8b9f20fe9d4a.png)
) (`georef_disabled` feature flag, false by default)
  - [x] disable street addresses if insufficient/no georeference quota set
  - [x] "_no georeferenced data on your data_" #4131

This scope also involves changes to the geocoded process (practically goes through the background importer) which are taken care of @xavijam separately, see PR #4159

### PR DESCRIPTION
**Pull request by viddo**
The last tab for the new georeference tab, fixes #4158 finally(!)

Some screenshots of various states, mostly some of the estimation variations + adding additional street addr columns/free texts for formatter:

screen shot 2015-07-06 at 18 28 29
screen shot 2015-07-06 at 18 29 42
screen shot 2015-07-06 at 18 25 27
screen shot 2015-07-06 at 18 25 39
screen shot 2015-07-06 at 18 25 44
screen shot 2015-07-06 at 18 26 13
screen shot 2015-07-06 at 18 27 39
screen shot 2015-07-06 at 18 32 01
screen shot 2015-07-06 at 18 45 08

### CODE REVIEW

**Comment by viddo** [comment]
please review @xavijam
cc @josecruz @saleiva for visuals
deployed on ded03 for testing

**Comment by viddo** [comment]
@rafatower a question for when I implement the quota information… I found [this piece](https://github.com/CartoDB/cartodb/blob/master/lib/assets/javascripts/cartodb/table/geocoding_dialog/panes/geocoding_dialog_pane_address.js#L15-L16), so currently the block price is hardcoded in the client-side, but looking at the `user.get('geocoding')` object it seems like a `block_price` param is available there - is it safe to remove this now and use the value from the user object instead?

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4734/))

**Comment by rafatower** [comment]
@viddo, answering with a question: what is `block_size` used for? at first I'd say it is better to use what comes from user object.

**Comment by viddo** [comment]
@rafatower to display information regarding possible costs after monthly credits are used and user has a "soft limit" set to true, see https://github.com/CartoDB/cartodb/blob/2b70ef52beef66e7ba27f5e266422924f86b38cb/lib/assets/javascripts/cartodb/table/views/geocoding_dialog/geocoding_dialog_pane_address.jst.ejs#L30-L40

**Comment by rafatower** [comment]
`block_size` <> `block_price` hence my confusion (not sure you spotted the difference).

`block_size` can be considered a constant that won't change over time. I don't know if it is sent on the user object. If it is, I'm ok with the change.

**Comment by josecruz** [comment]
Nice work! Loving it :D

Just to remember: I made some notes about the geocode: #4262

**Comment by viddo** [comment]
@josecruz Yep I know, but I'll deal with those separately (want to keep the scope focused on the street tab in this PR)

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4744/))

**Comment by viddo** [comment]
@rafatower ah yes, you're right! That value is not sent on the user object (or other), so need to set it like this for now then. Thx for spotting that detail. :+1:

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4759/))

File: app/assets/stylesheets/common/dialog.css.scss [hunk 1]
```diff
@@ -120,6 +120,26 @@ body.is-inDialog {
   @include display-flex();
   @include justify-content(center);
 }
+.Dialog-preFooter {
```
Comment (xavijam) [inline app/assets/stylesheets/common/dialog.css.scss]: preprepreFooter :DDD, ayyy! I know the problem, no more words.

File: lib/assets/javascripts/cartodb/common/dialogs/georeference/georeference_model.js [hunk 2]
```diff
@@ -179,19 +186,27 @@ module.exports = cdb.core.Model.extend({
   },
 
   _maybeDisabledStreetAddresses: function() {
-    var user = this.get('user');
-
     // Credits are only used for users that use non-google geocoding (i.e. the default use-case)
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/common/dialogs/georeference/georeference_model.js]: Are you taking into account the possibility of having the high resolution disabled (on-premise) or open source users?

File: lib/assets/javascripts/cartodb/common/dialogs/georeference/street_addresses/estimation_view.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+var _ = require('underscore');
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/common/dialogs/georeference/street_addresses/estimation_view.js]: If the estimation cost fails, what would it happen?

**Comment by xavijam** [comment]
Just those two questions and :us:.

File: app/assets/stylesheets/common/dialog.css.scss (hunk 1 above)
Comment (viddo) [inline app/assets/stylesheets/common/dialog.css.scss]: Yep, I have some idea how to deal with it but I think it's better to deal with separately, e.g. when fixing the headers in #4175

File: lib/assets/javascripts/cartodb/common/dialogs/georeference/georeference_model.js (hunk 2 above)
Comment (viddo) [inline lib/assets/javascripts/cartodb/common/dialogs/georeference/georeference_model.js]: Yeah, I followed the logic from old modal, but it was implemented in the last PR: https://github.com/CartoDB/cartodb/pull/4255/files#diff-ed6b35418b07db7f04fac9918463903cR165

File: lib/assets/javascripts/cartodb/common/dialogs/georeference/street_addresses/estimation_view.js (hunk 3 above)
Comment (viddo) [inline lib/assets/javascripts/cartodb/common/dialogs/georeference/street_addresses/estimation_view.js]: screen shot 2015-07-07 at 17 12 33 thought I had it covered but missed it I see now, so thanks. Solved it in 2181f06, like how it's done in the old modal.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4771/))

**Comment by viddo** [comment]
The missing quota info will be solved separately since not ready yet… see #4324

**Pull request merged**